### PR TITLE
using CompletableFuture instead of Callable for asynchronous controller in Spring Boot

### PR DESCRIPTION
**Pull request by thebignet**

### CODE REVIEW

**Comment by wing328** [comment]
@thebignet thanks for the PR on Spring Boot. I've just merged https://github.com/swagger-api/swagger-codegen/pull/3133 into master

Please pull the latest master and rebase the change accordingly.

**Comment by cbornet** [comment]
I think you can't use CompletableFuture on java7...

**Comment by thebignet** [comment]
You're right.How about changing the async property to accept java7 or java8 as a value like what is done for dateLibrary ?

File: modules/swagger-codegen/src/main/resources/JavaSpring/api.mustache [hunk 1]
```diff
@@ -43,10 +43,11 @@ public interface {{classname}} {
...
         consumes = { {{#consumes}}"{{{mediaType}}}"{{#hasMore}}, {{/hasMore}}{{/consumes}} },{{/hasConsumes}}{{/singleContentTypes}}
         method = RequestMethod.{{httpMethod}})
-    {{#java8}}default {{/java8}}{{#async}}Callable<{{/async}}ResponseEntity<{{>returnTypes}}>{{#async}}>{{/async}} {{operationId}}({{#allParams}}{{>queryParams}}{{>pathParams}}{{>headerParams}}{{>bodyParams}}{{>formParams}}{{#hasMore}},
+    {{#java8}}default {{/java8}}{{#async}}{{^java8}}Callable{{/java8}}{{#java8}}CompletableFuture
+            <{{/java8}}{{/async}}ResponseEntity<{{>returnTypes}}>{{#async}}>{{/async}} {{operationId}}({{#allParams}}{{>queryParams}}{{>pathParams}}{{>headerParams}}{{>bodyParams}}{{>formParams}}{{#hasMore}},
         {{/hasMore}}{{/allParams}}){{^java8}};{{/java8}}{{#java8}} {
         // do some magic!
-        return {{#async}}() -> {{/async}}new ResponseEntity<{{>returnTypes}}>(HttpStatus.OK);
+    return {{#async}}{{^java8}}() -> {{/java8}}{{#java8}}CompletableFuture.completedFuture({{/java8}}{{/async}}new ResponseEntity<{{>returnTypes}}>(HttpStatus.OK){{#async}}){{/async}};
```
Comment (cbornet) [inline modules/swagger-codegen/src/main/resources/JavaSpring/api.mustache]: It's a java8 block so ```{{^java8}}() -> {{/java8}}``` is not needed

File: modules/swagger-codegen/src/main/resources/JavaSpring/apiController.mustache [hunk 2]
```diff
@@ -20,24 +20,27 @@ import org.springframework.web.multipart.MultipartFile;
...
+{{/async}}
+{{/java8}}
 
 {{>generatedAnnotation}}
 @Controller
 {{#operations}}
 public class {{classname}}Controller implements {{classname}} {
 {{^java8}}{{#operation}}
-    public {{#async}}Callable<{{/async}}ResponseEntity<{{>returnTypes}}>{{#async}}>{{/async}} {{operationId}}({{#allParams}}{{>queryParams}}{{>pathParams}}{{>headerParams}}{{>bodyParams}}{{>formParams}}{{#hasMore}},
+    public {{#async}}{{^java8}}Callable{{/java8}}{{#java8}}CompletableFuture{{/java8}}
```
Comment (cbornet) [inline modules/swagger-codegen/src/main/resources/JavaSpring/apiController.mustache]: It's a "non-java8" block so no need to put java8 code here. Java8 default implementation is in the interface itself with default keyword

**Comment by cbornet** [comment]
LGTM. Maybe you can update the java8 spring sample ?

**Comment by thebignet** [comment]
Java8 async sample is updated.  I used a separate commit, but feel free to squash the branch commit before merging

**Comment by wing328** [comment]
@thebignet thanks for the PR. I've merged it into master.

**Pull request merged**